tests/containers/podman_quadlet.pm

```diff
@@ -98,30 +97,32 @@ sub run {
     record_info('Unit', script_output("$quadlet -v -dryrun"));
 
     # check that services are not present yet
-    check_unit_states('not-found');
+    for my $unit (@units) {
+        check_unit_enabled($unit, 'not-found');
+    }
 
     # start the generator and check whether the files are generated
     systemctl("daemon-reload");
-    check_unit_states();
     for my $unit (@units) {
+        check_unit_enabled($unit);
         systemctl("is-active $unit", expect_false => 1);
     }
 
     # start build unit
-    script_retry("podman pull $src_image", retry => 3, delay => 60, timeout => 180);
-    systemctl("start $unit_name-build.service", timeout => 180);
-    record_info('Build output', script_output("journalctl --no-pager -u $unit_name-build"));
-
-    systemctl("is-active $unit_name-build.service", expect_false => 1);
-    validate_script_output('podman images -n', qr/$build_imagetag/);
+    if (check_min_runtime_version('5.2.0')) {
+        script_retry("podman pull $src_image", retry => 3, delay => 60, timeout => 180);
+        systemctl("start $unit_name-build.service", timeout => 180);
+        record_info('Build output', script_output("journalctl --no-pager -u $unit_name-build"));
+        systemctl("is-active $unit_name-build.service", expect_false => 1);
```

Comment:
We need to check if this unit has run, not if it is active. This check will always pass.

Reply:
But after starting it wouldn't be active if there was an error. Also, this is what the test had before, I only added the if statement.

Reply:
Ah ok, then let's leave it for now.